rosbag2_transport/src/rosbag2_transport/qos.cpp

```diff
@@ -68,3 +68,35 @@ bool convert<rosbag2_transport::Rosbag2QoS>::decode(
   return true;
 }
 }  // namespace YAML
+
+
+namespace
+{
+/// Check if two rmw_time_t have the same values.
+bool operator==(const rmw_time_t & left, const rmw_time_t & right)
+{
+  return left.sec == right.sec && left.nsec == right.nsec;
+}
+}  // unnamed namespace
+
+
+namespace rosbag2_transport
+{
+// TODO(emersonknapp) it is the long term goal to have similar logic exist within rmw/rcl/rclcpp,
+//  but it's not yet determined where that logic will go. We should remove this function before
+//  G-Turtle
+bool Rosbag2QoS::compatibility_policies_exactly_equal(const rclcpp::QoS & other) const
```

Comment:
Does it make sense to have this as an `==` operator instead?

Reply:
I think it's too specific to use ==, I don't want anyone accidentally thinking it's full equality